TMEDIA-788 : Style Components for Commerce Theme

ecarlisle opened the pull request:
## Ticket

- [TMEDIA-788 : Style the components for Commerce theme](https://arcpublishing.atlassian.net/browse/TMEDIA-788)

## Description

Global and alias tokens were used to update the following most Commerce components. Some components like Image, Grid, Separator, and Video did not require styling.

## Acceptance Criteria

- Use the latest agreed-upon global token and alias definitions
- Update the default styles for each component
- Update Storybook

## Test Steps

1. Checkout branch - `git checkout TMEDIA-788-style-commerce-theme`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Compare component styles to [Figma specifications](https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?node-id=3505%3A28819). Components are styled with a combination of [global-tokens](https://github.com/WPMedia/arc-themes-components/blob/TMEDIA-788-style-commerce-theme/.storybook/themes/commerce.scss#L6-L161) and [alias-tokens specified in Confluence](https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3118465081/Design+Tokens).

### Testing Notes:
1. Pill uses a `border-radius` of `2em`. Implementation for a respective token has been deferred for future scope.
2. Input uses a 2px outline-offset, another value not available in tokens.
3. All components using typographic alias tokens will use both the respective `font-size` and `line-height`. For example, if a component has a text styling of "Small" in Figma, it should have the following CSS properties:
```
  font-size: var(--body-font-size-small),
  line-height: var(--body-line-height-small),
```
In Figma, components are currently shown using the `desktop` size of these properties.
<img width="519" alt="image" src="https://user-images.githubusercontent.com/452134/167478148-40c86513-3b5d-4641-9ae9-7d84e57e0d75.png">

You can use the following tables for the 3 typography alias token values:

`body-font-size`
|    default    |      |    desktop    |      |
|:-------------:|:----:|:-------------:|:----:|
| font-size-3   | 14px | font-size-4   | 16px |
| line-height-3 | 20px | line-height-3 | 24px |

`body-font-size-small`
|    default    |      |    desktop    |      |
|:-------------:|:----:|:-------------:|:----:|
| font-size-2   | 12px | font-size-3   | 14px |
| line-height-2 | 16px | line-height-3 | 20px |

`body-font-size-tiny`
|    default    |      |    desktop    |      |
|:-------------:|:----:|:-------------:|:----:|
| font-size-1   | 10px | font-size-2   | 12px |
| line-height-1 | 14px | line-height-2 | 16px |

5. The `button` component uses a bold font (`global-font-weight-7`) despite other typographic aliases. 
6. The `pill` component breaks convention from its labeled size in Figma. It will use `global-font-size-2` and `global-line-height-2` at all breakpoints.

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed relevant documentation has been updated/added.
- [ ] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

ecarlisle committed:
Update Storybook head w/ Google Fonts.

ecarlisle committed:
Update style tokens for Commerce components.

matthewroach committed:
Update media item to address issues

ecarlisle committed:
Updates for pill and Stack divider.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -209,15 +209,21 @@
+				attribution: (
+					color: var(--text-color),
+					font-size: var(--body-font-size-small),
+					font-weight: var(--global-font-weight-7),
```
```suggestion
					font-weight: var(--body-font-weight-small),
```

looks like this should be 400 based on mocks. this goes with the alias 400 for font weight small and font-size small. small thing -- sorry

JackHowa replied:
<img width="1094" alt="Screen Shot 2022-05-06 at 10 22 05" src="https://user-images.githubusercontent.com/5950956/167163136-f1c45867-b4bc-40ed-838f-d5d04052ca7b.png">

ecarlisle replied:
@JackHowa : Thanks, corrected! And no worry about the size of change. I want to be on par w/ Design specs 100%!

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -209,15 +209,21 @@
 			components: (
+				attribution: (
+					color: var(--text-color),
+					font-size: var(--body-font-size-small),
```
Is this attribution meant to scale down to 12px on mobile? I'm not sure based on the designs

Looks like it's always 14px in the designs. 
<img width="883" alt="Screen Shot 2022-05-06 at 11 00 20" src="https://user-images.githubusercontent.com/5950956/167169680-b75cdb5a-1f4c-4270-b1be-62a64d87824c.png">
 
on mobile 

<img width="150" alt="Screen Shot 2022-05-06 at 10 59 23" src="https://user-images.githubusercontent.com/5950956/167169710-7ad7b26a-dc13-468f-baf4-456f8825c2b7.png">

ecarlisle replied:
I'm going to discuss this w/ Kelly very shortly
I may need to change both Attribution and Date.
Will follow-up! Thanks!

RobinGiannattasio replied:
Hey! So let's make sure to bring this up in design sync on Monday.

It seems like we do not have have figma illustrations for all mobile vs desktop designs yet, which is cool, but without them we don't know when the components should vary based on breakpoint or when they should be the same. In lieu of that, we have two options:

- Assume mobile and desktop are the same unless explicitly illustrated in figma. This means in Figma we would paying more attention to the typography *size and line-height* attributes instead of the typography *AG* key which denotes a token. In code, we would ignore our body responsive alias tokens in favor of explicit global tokens. 
- **_Assume the we should be using responsive values unless explicitly illustrated in figma. This means in Figma we would paying more attention to the typography *AG* key which denotes a token instead of the typography *size and line-height* attributes. In terms of code in this case we should use our alias tokens instead of global tokens._** <- we will do this (see follow up comment)

Which direction should we go? Either way, we might need a desktop specific follow up ticket to do any clean up afterwards.

RobinGiannattasio replied:
Design Sync Update:

- We should assume that we should use the responsive alias tokens unless otherwise noted
- The pill right now is the only one that we know should be the same across breakpoints
- Any other component that needs to be updated because it did not fit into our assumption will be captured in the scope of a separate ticket after design has a chance to review.

ecarlisle replied:
@JackHowa : Following up.  As per design sync meeting and Robin's notes above, the Attribution will use global tokens w/ its design displayed in Figma using the desktop metrics.

ecarlisle committed:
Change attribution to 400 font weight.

ecarlisle committed:
Token changes:

- Set line height of body-font-size-small to global-line-height-4 (24px)
- All components w/ typographic aliases use the same font-size and line-height.
- Remove some properties not needed as they're the same as CSS cascade.

ecarlisle committed:
Have pill use global tokens for typography.

ecarlisle commented on .storybook/themes/commerce.scss:
```diff
@@ -301,6 +313,102 @@
+					border-radius: 2em,
+					color: var(--global-white),
+					font-size: var(--global-font-size-2),
+					line-height: var(--global-line-height-2),
```
**Note:** `pill` is an exception to its typographic specs in Figma. It will use global tokens for `font-size` and `line-height`. Typography will be the same on all breakpoints.

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -301,6 +313,102 @@
+				),
+				stack-divider-vertical: (
+					border-bottom: var(--global-border-width-1) var(--global-border-style-1)
+						var(--form-border-color),
```
The use of `--form-border-color` seems wrong in this instance when we have `border-color` alias

ecarlisle replied:
Good call! Replaced.

matthewroach replied:
@ecarlisle Did you push the change?

ecarlisle replied:
@matthewroach : Sorry, that's on me. Updated the base branch and pushed. Must not have had my latest commits grouped in. Changes are pushed.

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -301,6 +313,102 @@
+				stack-divider-horizontal: (
+					border: none,
+					border-left: var(--global-border-width-1) var(--global-border-style-1)
+						var(--form-border-color),
```
The use of `--form-border-color` seems wrong in this instance when we have `border-color` alias

ecarlisle replied:
Got this one too. Replaced.

matthewroach replied:
@ecarlisle Did you push the change?

ecarlisle replied:
@matthewroach : See above. Thank you for asking!

ecarlisle committed:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-788-style-commerce-theme

ecarlisle committed:
Use border-color alias on stack dividers.

ecarlisle committed:
Add Badge component, styles, stories, and tests.

ecarlisle committed:
Add badge styles to Commerce theme.

ecarlisle committed:
Add Badge to alphabetized component object file.

ecarlisle committed:
Add badge styles to main Sass file.

ecarlisle committed:
Correct aliases to Badge background colors to those indicated in Figma.

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -301,6 +313,101 @@
+				media-item-title: (
+					font-weight: var(--global-font-weight-7),
+				),
+				overline: (
```
We need to account for Overline link to follow the same

<img width="382" alt="Google Chrome-2022-05-10-20-51 52" src="https://user-images.githubusercontent.com/868127/167710803-73cedb9a-97e4-4c95-ae63-40d436ae40cc.png">

ecarlisle replied:
@matthewroach : To confirm, do we want the Overline Link to have the same color and not to be underlined?
For example, adding the following:
```
overline: (
  color: var(--text-color),
  text-decoration: none,
  ...
),
```

Thanks!

matthewroach replied:
At minimum yes

ecarlisle replied:
Thanks! I made the change above and am seeing the following in Storybook. The only difference I'm seeing is the cursor change when hovering over the Overline link.

<img width="450" alt="image" src="https://user-images.githubusercontent.com/452134/167713939-1f25ed7c-be95-4949-903b-40b12fa0c9bf.png">

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -301,6 +313,101 @@
+				media-item-title: (
+					font-weight: var(--global-font-weight-7),
+				),
+				overline: (
```
We need to account for Overline link to follow the same

<img width="382" alt="Google Chrome-2022-05-10-20-51 52" src="https://user-images.githubusercontent.com/868127/167710803-73cedb9a-97e4-4c95-ae63-40d436ae40cc.png">

ecarlisle committed:
Correct alias color mappings for danger and success.

ecarlisle committed:
Style Overline link to be identical to Overline.

ecarlisle committed:
Update src/components/badge/index.test.jsx

Use `BADGE_TEXT` rather than string.

Co-authored-by: Matthew Roach <matthewroach@users.noreply.github.com>

ecarlisle committed:
Update src/components/badge/index.test.jsx

Use BADGE_TEXT const instead of text equivalent.

Co-authored-by: Matthew Roach <matthewroach@users.noreply.github.com>

ecarlisle committed:
Update src/components/badge/index.test.jsx

Use BADGE_TEXT const instead of text equivalent.

Co-authored-by: Matthew Roach <matthewroach@users.noreply.github.com>

ecarlisle committed:
Update src/components/badge/index.test.jsx

Use BADGE_TEXT const instead of text equivalent.

Co-authored-by: Matthew Roach <matthewroach@users.noreply.github.com>

ecarlisle committed:
Update src/components/badge/index.test.jsx

Use BADGE_TEXT const instead of text equivalent.

Co-authored-by: Matthew Roach <matthewroach@users.noreply.github.com>

ecarlisle committed:
Merge pull request #93 from WPMedia/TMEDIA-851-component-badge

TMEDIA-851 : Component - Badge

matthewroach commented:
@ecarlisle Once this is merged don't forget to create a pull request against the feature pack to ensure we are up to date with the source of truth
